### PR TITLE
fixed some covs

### PR DESCRIPTION
**Pull request by Vane11ope**

### CODE REVIEW

File: libr/io/p/io_gprobe.c [hunk 1]
```diff
@@ -711,6 +711,8 @@ static int gprobe_read (struct gport *port, ut32 addr, ut8 *buf, ut32 count) {
 	int res;
 
 	if (!request || !reply) {
+		r_buf_free (request);
```
Comment (XVilka) [inline libr/io/p/io_gprobe.c]: This is error-prone, since you free both pointers even if one of them can be NULL.

File: libr/io/p/io_gprobe.c [hunk 2]
```diff
@@ -754,6 +756,8 @@ static int gprobe_write (struct gport *port, ut32 addr, const ut8 *buf, ut32 cou
 	ut8 count_be[4];
 
 	if (!request || !reply) {
+		r_buf_free (request);
```
Comment (XVilka) [inline libr/io/p/io_gprobe.c]: Same here.

**Comment by Vane11ope** [comment]
ah right fixed!

**Comment by Vane11ope** [comment]
and this is red again but seeing from log i reckon that's not because of this

File: libr/io/p/io_gprobe.c [hunk 3]
```diff
@@ -747,13 +752,18 @@ static int gprobe_read (struct gport *port, ut32 addr, ut8 *buf, ut32 count) {
...
 
-	if (!request || !reply) {
+	RBuffer *request = r_buf_new ();
+	if (!request) {
+		r_buf_free (request);
+		return -1;
+	}
+	RBuffer *reply = r_buf_new ();
+	if (!reply) {
+		r_buf_free (reply);
```
Comment (radare) [inline libr/io/p/io_gprobe.c]: free the request here

File: libr/io/p/io_gprobe.c [hunk 4]
```diff
@@ -747,13 +752,18 @@ static int gprobe_read (struct gport *port, ut32 addr, ut8 *buf, ut32 count) {
...
 
 static int gprobe_write (struct gport *port, ut32 addr, const ut8 *buf, ut32 count) {
-	RBuffer *request = r_buf_new ();
-	RBuffer *reply = r_buf_new ();
 	const ut8 cmd = GPROBE_RAM_WRITE_2;
 	ut8 addr_be[4];
 	ut8 count_be[4];
 
-	if (!request || !reply) {
+	RBuffer *request = r_buf_new ();
```
Comment (radare) [inline libr/io/p/io_gprobe.c]: you can keep the same code and free both and quit in a single conditional block if you want. will also be correct

File: libr/io/p/io_gprobe.c [hunk 5]
```diff
@@ -703,14 +703,19 @@ static int gprobe_send_request_sp (struct gport *port, RBuffer *request) {
...
+	if (!request) {
+		r_buf_free (request);
+		return -1;
+	}
+	RBuffer *reply = r_buf_new ();
+	if (!reply) {
+		r_buf_free (reply);
```
Comment (radare) [inline libr/io/p/io_gprobe.c]: free request here.

**Comment by radare** [comment]
pls rebase . hopefully master will be green

**Comment by XVilka** [comment]
Sorry, it is my suggestion, seems free(NULL) is not problematic anymore on any platform: https://stackoverflow.com/questions/1938735/does-freeptr-where-ptr-is-null-corrupt-memory

**Comment by radare** [comment]
its never been a problem to do free(NULL) on any system, ever

**Comment by Vane11ope** [comment]
so how about now

**Comment by XVilka** [comment]
@radare it was - check the link please.

**Pull request merged**